### PR TITLE
feat: protocol(22) support and upstream changes

### LINKED ISSUE #257: Error running a mainnet fork in docker
📎 Log Output
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseError(Error("unknown variant `Version20`, expected one of `Version0`, `Version1`, `Version2`, `Version3`, `Version4`, `Version5`, `Version6`, `Version7`, `Version8`, `Version9`, `Version10`, `Version11`, `Version12`, `Version13`, `Version14`, `Version15`, `Version16`, `Version17`, `Version18`, `Version19`", line: 1, column: 643))', src/fork.rs:359:14
```

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
* Updates zksync-era dep to latest commit `cd748e9593949222fe6516fc18c987b6cdf39882` 
* Updates protocol version to support 21, 22
* Updates contracts to use submodules from #267 

# Why :hand:
* Keeps deps up to date and provides support for latest protocol upgrades
* Fixes forking from mainnet 
* Improves the updating flow of system contract deps

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

Screenshot 2024-03-13 at 10 00 23 AM
Screenshot 2024-03-13 at 10 00 46 AM

Closes #257

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -232,22 +232,19 @@ pub fn utc_datetime_from_epoch_ms(millis: u64) -> DateTime<Utc> {
...
             | Web3Error::LogsLimitExceeded(_, _, _)
-            | Web3Error::InvalidFilterBlockHash => ErrorCode::InvalidParams,
-            Web3Error::SubmitTransactionError(_, _) | Web3Error::SerializationError(_) => 3.into(),
-            Web3Error::PubSubTimeout => 4.into(),
-            Web3Error::RequestTimeout => 5.into(),
-            Web3Error::TreeApiUnavailable => 6.into(),
+            | Web3Error::InvalidFilterBlockHash
+            | Web3Error::TreeApiUnavailable => ErrorCode::InvalidParams,
+            Web3Error::SubmitTransactionError(_, _) | Web3Error::SerializationError(_) => {
+                ErrorCode::ServerError(3)
```
Comment (MexicanAce) [inline src/utils.rs:246]: Where does the `3` come from?

Comment (dutterbutter) [reply src/utils.rs:246]: Truly not sure lol we were using it previously (e.g. `3.into()`). It puts it into the `JSONRPC error code` enum as expected. Happy to change but kept it as it was done previously.

Comment (MexicanAce) [reply src/utils.rs:246]: Ah yup, let's keep as-is 😅

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**